examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,146 @@
-
-The histogram of the estimated weights is very peaked, as a sparsity-inducing
-prior is implied on the weights.
-
-The estimation of the model is done by iteratively maximizing the
+The estimation of the model is done in both cases by iteratively maximizing the
 marginal log-likelihood of the observations.
 
-We also plot predictions and uncertainties for ARD
-for one dimensional regression using polynomial feature expansion.
-Note the uncertainty starts going up on the right side of the plot.
-This is because these test samples are outside of the range of the training
-samples.
+We also plot predictions and uncertainties for ARD and Bayesian Ridge
+regressions using a polynomial feature expansion to fit a non-linear feature.
+Notice that the ARD regression captures the ground truth the best, but due to
+the limitations of a polynomial regression, both models fail when extrapolating.
 
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from scipy import stats
-
-from sklearn.linear_model import ARDRegression, LinearRegression
-
-# %%
-# Generating simulated data with Gaussian weights
-
-# Parameters of the example
-np.random.seed(0)
-n_samples, n_features = 100, 100
-# Create Gaussian data
-X = np.random.randn(n_samples, n_features)
-# Create weights with a precision lambda_ of 4.
-lambda_ = 4.0
-w = np.zeros(n_features)
-# Only keep 10 weights of interest
-relevant_features = np.random.randint(0, n_features, 10)
-for i in relevant_features:
-    w[i] = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(lambda_))
-# Create noise with a precision alpha of 50.
-alpha_ = 50.0
-noise = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(alpha_), size=n_samples)
-# Create the target
-y = np.dot(X, w) + noise
-
 # %%
-# Fit the ARD Regression
-clf = ARDRegression(compute_score=True)
-clf.fit(X, y)
-
-ols = LinearRegression()
-ols.fit(X, y)
+# Generate synthetic dataset
+# --------------------------
+from sklearn.datasets import make_regression
+
+X, y, w = make_regression(
+    n_samples=100,
+    n_features=100,
+    n_informative=10,
+    noise=8,
+    coef=True,
+    random_state=42,
+)
 
 # %%
-# Plot the true weights, the estimated weights, the histogram of the
-# weights, and predictions with standard deviations
-plt.figure(figsize=(6, 5))
-plt.title("Weights of the model")
-plt.plot(clf.coef_, color="darkblue", linestyle="-", linewidth=2, label="ARD estimate")
-plt.plot(
-    ols.coef_, color="yellowgreen", linestyle=":", linewidth=2, label="OLS estimate"
+# Fit the regressions
+# -------------------
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression()
+brr = BayesianRidge(compute_score=True, n_iter=30)
+ard = ARDRegression(compute_score=True, n_iter=30)
+
+olr.fit(X, y)
+brr.fit(X, y)
+ard.fit(X, y)
+
+df = pd.DataFrame(
+    {
+        "LinearRegression": olr.coef_,
+        "BayesianRidge": brr.coef_,
+        "ARDRegression": ard.coef_,
+        "true weights": w,
```

Comment:
"Weights" can refer to many weighting (class weights, samples weights, ...).

I think for clarity, it is better to use "coefficients" as a wording for this example.

```suggestion
        "true coefficients": w,
```

This is the case here, but also in other places. WDYT?